authors/apps/authentication/test/test_verification.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+from django.urls import reverse
+from django.core import mail
+from rest_framework.test import APITestCase, APIRequestFactory
+from rest_framework import status
+
+from ..models import User
+from ..views import RegistrationAPIView, VerifyAccount
+
+
+class VerificationTestCase(APITestCase):
+    def setUp(self):
+        self.data = {
+            "user": {
+                "username": "Ntale",
+                "email": "shadik.ntale@andela.com",
+                "password": "AthenaD0a"
+            }
+        }
+        self.url = reverse('registration')
+        self.client.post(self.url, self.data, format='json')
+        self.request = APIRequestFactory().post(
+            reverse("registration"))
+
+    def verify_account(self, token, uidb64):
+        request = APIRequestFactory().get(
+            reverse(
+                "activate_account",
+                kwargs={
+                    "token": token,
+                    "uidb64": uidb64}))
+        verify_account = VerifyAccount.as_view()
+        response = verify_account(request, token=token, uidb64=uidb64)
+        return response
+
+    def test_sends_email(self):
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertEqual(
+            mail.outbox[0].subject,
+            "Author's Heaven account email verification")
+
+    def test_account_verified(self):
+        user = User.objects.get()
+        req = self.request
+        token, uidb64 = RegistrationAPIView.generate_activation_link(
+            user, req, send=False)
+        response = self.verify_account(token, uidb64)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        user = User.objects.get()
+        self.assertTrue(user.is_verified)
+
+    def test_generate_activation_link_function(self):
```

Comment:
Assert for whether `send_mail` is called on success.

Reply:
@collin5 I asserted that. In the setUp, I signed up a user and on signup, an email had to be sent so the outbox will then be 1. When I go ahead and call the `generate_activation_link` function without inputting the `*kwarg` send, the outbox will then be 2 because the `send_mail` function will be called. When I set the `kwarg` send to false though, the `send_mail` function will not be called so the outbox will remain 1. Then I `assertFalse` that the outbox will be 2.